### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by quentinlesceller**
Replace https://github.com/mimblewimble/grin/pull/2742 and fix https://github.com/mimblewimble/grin/issues/2691. 

While GitLab is a great candidate for CI, it does not support macOS and Windows builds with the SaaS solution.

This PR adds Azure Pipelines as Grin CI.

Pros:
- Windows, Linux and macOS builds.
- Details on each job directly on Github.
- [Unlimited CI/CD minutes](https://azure.microsoft.com/en-us/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/).

Cons:
- No caching yet (but it is in the [pipeline](https://devblogs.microsoft.com/devops/adding-caching-to-azure-pipelines/)).
- Microsoft.
- Documentation scattered and not always super clear.

To do:
Complete release process with Azure Pipelines.

Example of build process here:
- Azure side: https://dev.azure.com/qlesceller/qlesceller/_build/results?buildId=49
- Github side: https://github.com/quentinlesceller/grin/pull/3

If you don't like the facts that there is 12 tasks running on GitHub I can always let it display as one task.

If this is accepted next steps would be to:
- @ignopeverell Remove GitLab integration and add Azure Pipelines on Grin repo.
- Add complete release process (which is doable https://nbsoftsolutions.com/blog/azure-pipelines-for-rust-projects).

### CODE REVIEW

**Comment by quentinlesceller** [comment]
I guess you need to create an Azure account. I'll let you know when to merge I'm trying to do the GitHub release process and once that is done we'll be able to merge.

**Comment by ignopeverell** [comment]
Where are we at with Azure CI? This PR needs some minor conflict fix.

**Comment by quentinlesceller** [comment]
@ignopeverell as you can see I needed some cleanup https://github.com/mimblewimble/grin/pull/2752. In order to debug Azure Pipeline CI. You can follow my progress [on GitHub](https://github.com/quentinlesceller/grin/pull/3) and on [Azure](https://dev.azure.com/qlesceller/qlesceller/_build/results?buildId=96&view=logs). Windows tests are still no passing but I'm making very good progress.
Fixing the merge conflict btw.

File: store/tests/pmmr.rs [hunk 1]
```diff
@@ -776,7 +776,7 @@ fn cleanup_rewind_files_test() {
 		&data_dir,
 		expected,
 		prefix_to_delete,
-		seconds_to_delete_after,
+		seconds_to_delete_after + 1,
```
Comment (quentinlesceller) [inline store/tests/pmmr.rs]: For those wondering why I had to do this:
This test create file such as they are removed if older than ```seconds_to_delete_after```. 
However this test is very fast so basically it does that.
1. Create file that are `seconds_to_delete_after` old.
2. Do some operation
3. Remove file that are older than `seconds_to_delete_after`.

However, on Windows CI (as you can see here https://dev.azure.com/qlesceller/qlesceller/_build/results?buildId=105&view=logs&jobId=75e8fe06-bfd6-5564-cb5c-dff4fd81606c&taskId=c33ab3a6-027f-5d3f-2c09-9b84aecf111c&lineStart=1608&lineEnd=1608&colStart=1&colEnd=9) this duration is in seconds thus the deletion never happens. 
```
let duration_since_accessed = now.duration_since(accessed)?;
  if duration_since_accessed <= cleanup_duration {
    return Ok(0); // these files are still too new
}
// Deletion happens here
```

Comment (antiochp) [reply store/tests/pmmr.rs]: 😑

**Comment by quentinlesceller** [comment]
Okay this is done here is an example of what a release looks like: https://github.com/quentinlesceller/grin/releases. 
The changeling is not as nice though (but it still can be generated manually if needed).
@ignopeverell I'll need the name of the connection and I'll update the yml file.
Once this is done we can merge and proceed to a test beta release @yeastplume. 
IMO we should keep Travis file around in case we want to switch back (but just disable the check).

**Pull request merged**